pkg/ovsdb/transact.go

```diff
@@ -0,0 +1,1094 @@
+	actual, ok := (*row)[c.Column].(string)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	fn := c.Function
+	expected, ok := c.Value.(string)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && actual != expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Condition) CompareUUID(row *map[string]interface{}) (bool, error) {
+	actual, ok := (*row)[c.Column].(libovsdb.UUID)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	fn := c.Function
+	expected, ok := c.Value.(libovsdb.UUID)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && actual != expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Condition) CompareSet(row *map[string]interface{}) (bool, error) {
+	actual, ok := (*row)[c.Column].(libovsdb.OvsSet)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	fn := c.Function
+	expected, ok := c.Value.(libovsdb.OvsSet)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && isEqualSet(actual, expected) {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && !isEqualSet(actual, expected) {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Condition) CompareMap(row *map[string]interface{}) (bool, error) {
+	actual, ok := (*row)[c.Column].(libovsdb.OvsMap)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	fn := c.Function
+	expected, ok := c.Value.(libovsdb.OvsMap)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && isEqualMap(actual, expected) {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && !isEqualMap(actual, expected) {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Condition) Compare(row *map[string]interface{}) (bool, error) {
+	switch (*row)[c.Column].(type) {
+	case int:
+		return c.CompareInteger(row)
+	case float64:
+		return c.CompareReal(row)
+	case bool:
+		return c.CompareBoolean(row)
+	case string:
+		return c.CompareString(row)
+	case libovsdb.OvsSet:
+		return c.CompareSet(row)
+	case libovsdb.OvsMap:
+		return c.CompareMap(row)
+	case libovsdb.UUID:
+		return c.CompareUUID(row)
+	default:
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+}
+
+func isCondTypeUUID(cond1 interface{}) (string, error) {
+	cond2, ok := cond1.([]interface{})
+	if !ok {
+		return "", errors.New(E_ERROR)
+	}
+	condition, err := NewCondition(cond2)
+	if err != nil {
+		return "", err
+	}
+	if condition.Column != COL_UUID {
+		return "", nil
+	}
+	if condition.Function != FN_EQ {
+		return "", nil
+	}
+	ovsUUID, ok := condition.Value.(libovsdb.UUID)
+	if !ok {
+		return "", errors.New(E_ERROR)
+	}
+	_, err = guuid.Parse(ovsUUID.GoUUID)
```

Comment:
can we reuse the logic from uuid.validateUUID()

Reply:
ok